history/history-network.md

```diff
@@ -236,45 +236,79 @@ content          = SSZ.serialize(block_header_with_proof)
+> Note: The history network does not provide a mechanism for knowing the HEAD of the chain. Clients to this network **MUST** use an external oracle for this information. The Portal Beacon Network is able to provide this information.
 
 > Note: The content-id for this data type is not meaningful.
 
-> Note: This message is not valid for Gossip.  Clients **SHOULD** not send or accept gossip messages for this content type.
+> Note: Clients **SHOULD** avoid requesting missing headers from the last 64 block non-finalized window if their external oracle has updated the tip of the chain in a way that invalidates previously valid headers, as it is expected that valid updates are already being propagated to fill this discrepancy. Since all other Portal clients relying on the Portal Beacon Network would have the same view of the network, if implementations did try to find this content, it would end up being a DOS vector.
+
+> Note: When the external oracle updates the tip of the chain causing previously valid headers to be invalidated, the client **SHOULD** not prune any non-finalized ephemeral headers until the node can rebuild a valid chain from latest finalized header to the HEAD of the chain.
 
 > Note: Clients **SHOULD** implement a mechanism to purge headers older than 8192 blocks from their content databases.
 
+##### Ephemeral Headers by `Find Content`
+
 ```python
 # Content and content key
 
-ephemeral_headers_key = Container(block_hash: Bytes32, ancestor_count: uint8)
+find_content_ephemeral_headers_key = Container(block_hash: Bytes32, ancestor_count: uint8)
 selector                 = 0x04
 
 BlockHeader              = ByteList[MAX_HEADER_LENGTH]
 ephemeral_header_payload = List(BlockHeader, limit=MAX_EPHEMERAL_HEADER_PAYLOAD)
 
 content                  = SSZ.serialize(ephemeral_header_payload)
-content_key              = selector + SSZ.serialize(ephemeral_headers_key)
+content_key              = selector + SSZ.serialize(find_content_ephemeral_headers_key)
 ```
 
-The `ephemeral_headers_key` encodes a request for headers anchored to the block
-hash indicated by `ephemeral_headers_key.block_hash`.  The
-`ephemeral_headers_key.ancestor_count` **MUST** be in the inclusive range
+The `find_content_ephemeral_headers_key` encodes a request for headers anchored to the block
+hash indicated by `find_content_ephemeral_headers_key.block_hash`. The
+`find_content_ephemeral_headers_key.ancestor_count` **MUST** be in the inclusive range
 0-255.
 
 The `ephemeral_header_payload` is an SSZ list of RLP encoded block header
-objects.  This object is subject to the following validity conditions.
+objects. This object is subject to the following validity conditions.
 
+* This content type **MUST** only be used for `Find Content` requests
```

Comment:
```suggestion
* This content type **MUST** only be used for `FindContent` requests
```

Reply:
I will do this for all other instances as well then